src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -946,7 +946,17 @@ public override void VisitLocalDeclarationStatement(LocalDeclarationStatementSyn
         public override void VisitVariableDeclarator(VariableDeclaratorSyntax node)
         {
             Visit(node.ArgumentList);
-            Visit(node.Initializer?.Value);
+
+            if (node.Initializer is { } initializer)
+            {
+                var oldEnclosing = _enclosing;
+                var localInProgressBinder = new LocalInProgressBinder(initializer, _enclosing);
```

Comment:
>new LocalInProgressBinder(initializer, _enclosing) [](http://example.com/codeflow?start=44&length=50)

Would it make sense to limit creation of the extra binder to `const` declarations?